find.lic

```diff
@@ -2,76 +2,109 @@
-    pilgrimage_trail = [4179, 4180, 4181, 4182, 4183, 4184, 4185, 4186, 4187, 4188, 4189, 4190, 4191, 4192, 4193, 4194]
+  def initialize
     arg_definitions = [
       [
-        { name: 'npc', regex: /\w+/, description: 'Unique name/noun of the NPC to find' },
-        { name: 'follow', regex: /follow/i, optional: true, description: 'Follow the NPC once it is found' }
+        { name: 'town', options: ['crossing', 'dirge', 'shard', 'boar'], optional: true, description: 'Town/area to search' },
+        { name: 'npc', regex: /^(?!follow)([\w\s\-']+)$/i, optional: true, description: 'Unique name/noun of the NPC to find' },
+        { name: 'follow', regex: /^follow$/i, optional: true, description: 'Follow the NPC once it is found' }
       ]
     ]
 
     args = parse_args(arg_definitions, true)
-    @npc = args.npc
-    @follow = args.follow
 
-    @rooms = case args.npc
-             when /ozursus/i
-               crossing_academy
-             when /shaman/i
-               dirge_town
-             when /theurgist/i
-               shard_town
-             when /monk/i
-               pilgrimage_trail
-             else
-               crossing_town
-             end
-  end
+    # Show help menu if no NPC specified.
+    if args.npc.nil? || args.npc.empty?
+      force_start_script('find', ['help'])
+      exit
+    end
 
-  def initialize
-    setup
+    area = get_town_and_region(args.town, args.npc)
+    rooms = get_rooms(area.town, area.region)
 
-    start_index = @rooms.index(Room.current.id) || 0
-    search(@rooms.rotate(start_index)) unless found?
+    if search?(rooms, args.npc) && args.follow
+      follow(args.npc)
+    end
+  end
 
-    follow if @follow
+  # What geographical area are we searching in?
+  # Use this data to look up the actual rooms to explore.
+  def get_town_and_region(town, npc)
+    town = format_town(town)
+    case npc
+    when /ozursus/i
+      town = 'Crossing' if town.nil? || town.empty?
+      region = 'asemath_academy' if town == 'Crossing'
+    when /shaman/i
+      town = 'Dirge' if town.nil? || town.empty?
+      region = 'rooms'
+    when /theurgist/i
+      town = 'Shard' if town.nil? || town.empty?
+      region = 'rooms'
+    when /monk/i
+      town = 'Boar Clan' if town.nil? || town.empty?
+      region = 'pilgrimage_trail' if town == 'Boar Clan'
+    end
+    town = 'Crossing' if town.nil? || town.empty?
+    region = 'rooms' if region.nil? || region.empty?
+    return OpenStruct.new(
+      :town => town,
+      :region => region
+    )
   end
 
-  def search(room_ids)
-    return if room_ids.any? do |room_id|
-      walk_to(room_id)
-      found?
+  # If no town provided then 'Crossing' is used as default.
+  # Anything that starts with 'boar' is expanded to 'Boar Clan'.
+  # Each word in the town is capitalized to match values in base-town.yaml.
+  def format_town(town)
+    town = '' if town.nil?
+    case town
+    when /^boar/i
+      town = 'Boar Clan'
```

Comment:
Use convenience to specify shorter, or one-word variants of towns/areas and the script will expand the values to correct lookup value in `base-towns.yaml`.

For now, **Boar Clan** is the only city with wandering NPC rooms that needs this help.

Others could be "River" for "Riverhaven" and "Theren" for "Therenoborough", as future examples.